### ISSUE TITLE
Issue with SPIFFS storage instructions (version 0.6)

### ISSUE THREAD

**Issue by candideu** [iffybooks/pocket-wifi-portal#2]
Thank you so much for creating this Zine! I can't remember how I stumbled upon it, but I went ahead and bought the Wemos D1 mini as soon as I found the zine. 😄 

Last week, I finally tested it out and followed the instructions, but ran into issues with "Captive portal approach 2: Use SPIFFS storage" (page 25 of the screen zine PDF). After downloading the [plugin](https://github.com/esp8266/arduino-esp8266fs-plugin/releases) (v. 0.5.0), unzipping the file, and moving the "ESP8266FS" folder to the `[Sketchbook location]/tools/` directory, I was unable to locate the "ESP8266 Sketch Data Upload" plugin in the Arduino IDE Tools menu.

After doing some digging, it looks like the issue has been reported in the plugin's repo:
- https://github.com/esp8266/arduino-esp8266fs-plugin/issues/99
- https://github.com/esp8266/arduino-esp8266fs-plugin/issues/96
- https://github.com/esp8266/arduino-esp8266fs-plugin/issues/93

[According to another user](https://github.com/esp8266/arduino-esp8266fs-plugin/issues/96#issuecomment-1505209323), the plugin only works with Arduino IDE version 1.8.5 or older.

[Another user](https://github.com/esp8266/arduino-esp8266fs-plugin/issues/96#issuecomment-1742334198) says the following:

> this is the upstream issue: [arduino/arduino-ide#58](https://github.com/arduino/arduino-ide/issues/58)
> 
> tl;dr
> 
>     * plugin framework has completely changed. vscode integration improves, java plugins are not grandfathered in.
> 
>     * all old gui integrated tools aren't working, but expressif will help fix it
> 
>     * spiffs is deprecated, too, one ought to use one of the two newer libraries (littlefs and fat)
> 
>     * there is a littlefs plugin https://github.com/earlephilhower/arduino-esp8266littlefs-plugin which also has a gui for arduino >=2. It could be a starting point.

It looks like https://github.com/earlephilhower/arduino-littlefs-upload may be a starting point for a replacement, but I'm not the most well-versed with microcontrollers, and wanted to know if your team was aware of the issue and had other solutions (besides using an older version of the IDE). 

I'll try the littlefs plugin to see if it works and report back...

**Comment by candideu** [comment]
I went back to the zine, and realized that it specifies using version 1.8.x of Arduino on page 5... 🤦 That's my bad!

In terms of using https://github.com/earlephilhower/arduino-littlefs-upload on version 2.2.1 of Arduino: it doesn't look like it works as an alternative. I got an endless redirection page:

![image](https://github.com/iffybooks/pocket-wifi-portal/assets/55474996/c3ac4671-573a-4e5a-b0f1-e13f6e74f231)

I'll keep this issue open in case there are other ideas and solutions...

**Comment by candideu** [comment]
Okay, after lots of searching around and testing, I think I found a solution for the Arduino IDE version 2.2.1 or higher!! 🎉🎉

Here are the instructions for pages 25 - 28 of the zine (screen version):

# Install the [LittleFS uploader plugin](https://github.com/earlephilhower/arduino-littlefs-upload)

Go to the Arduino IDE plugins folder:

- Windows: `C:\Users\<username>\.arduinoIDE\plugins\`
- MacOS and Linux: `~/.arduinoIDE/plugins/`

You may need to create the "plugins" folder.

Next, download the vsix file to the "plugins" folder on your PC: https://github.com/earlephilhower/arduino-littlefs-upload/releases

![image](https://github.com/iffybooks/pocket-wifi-portal/assets/55474996/2192059a-7ace-4baa-a21e-6e289d9bf658)

Restart the Arduino IDE.

# Convert the `.ino` Arduino sketch file from SPIFFS to LittleFS

If you're using code that uses SPIFFS, you'll need to replace all mentions of SPIFFS in the `.ino` file with LittleFS (see this commit diff for an example of how I changed "Free Reading Wifi" project: https://github.com/candideu/pocket-wifi-portal-LittleFS/commit/b86479bb79717ef27a58b3134e4bd8be07d05758)

![image](https://github.com/iffybooks/pocket-wifi-portal/assets/55474996/7df11b21-c7a8-4bac-af78-6f5c6501a357)

Here's the converted/ported "Free Reading Wifi" file: https://github.com/candideu/pocket-wifi-portal-LittleFS/blob/main/04_Captive_portal_approach_2/Free_Reading_Wifi/Free_Reading_Wifi.ino

# Upload files with plugin

Place your project folder in your Sketchbook, as outlined in page 26 of the zine (screen version). 

Open the .ino file. In the Arduino IDE, hit `[Ctrl]` + `[Shift]` + `[P]` (Windows) or `[⌘]` + `[Shift]` + `[P]` (macOS), then type "Upload LittleFS to Pico/ESP8266". Select the option.

![image](https://github.com/iffybooks/pocket-wifi-portal/assets/55474996/0bb603ee-872b-4808-99bf-c5d19f35bb86)

When the upload is finished, you'll see the message **LittleFS upload completed!** beneath the code editor.

![image](https://github.com/iffybooks/pocket-wifi-portal/assets/55474996/5e32a3c2-b021-4f12-a840-7700eda5a2d8)

Click the **Upload** button in the top left to compile and upload the project code to the Wemos D1 mini.

![image](https://github.com/iffybooks/pocket-wifi-portal/assets/55474996/ab5e24e9-0724-4e82-aa2e-e38f7c3602a1)

When the upload is done, you'll see **Done uploading** beneath the code editor.

![image](https://github.com/iffybooks/pocket-wifi-portal/assets/55474996/c4b72a61-c99e-430c-b139-5e7484ac02d3)

**Comment by kabumos** [comment]
> Okay, after lots of searching around and testing, I think I found a solution for the Arduino IDE version 2.2.1 or higher!! 🎉🎉
> 
> Here are the instructions for pages 25 - 28 of the zine (screen version):
> 
> # Install the [LittleFS uploader plugin](https://github.com/earlephilhower/arduino-littlefs-upload)
> Go to the Arduino IDE plugins folder:
> 
> * Windows: `C:\Users\<username>\.arduinoIDE\plugins\`
> * MacOS and Linux: `~/.arduinoIDE/plugins/`
> 
> You may need to create the "plugins" folder.
> 
> Next, download the vsix file to the "plugins" folder on your PC: https://github.com/earlephilhower/arduino-littlefs-upload/releases
> 
> ![image](https://private-user-images.githubusercontent.com/55474996/287573759-2192059a-7ace-4baa-a21e-6e289d9bf658.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1NzczNTIsIm5iZiI6MTcwNTU3NzA1MiwicGF0aCI6Ii81NTQ3NDk5Ni8yODc1NzM3NTktMjE5MjA1OWEtN2FjZS00YmFhLWEyMWUtNmUyODlkOWJmNjU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE4VDExMjQxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ1ZmMyMWRhNGY2ZDkzZDQ2NTU2YmY5MjU4ZDViYjE2OWMxYjI5MWYwZmE3NmU1ZDMyMmMxYTM0Yjk5MTVjZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.19zoGDlFtJDkAxQ-I8YkBQS_OBtBc6HfjtxhkcSA728)
> 
> Restart the Arduino IDE.
> 
> # Convert the `.ino` Arduino sketch file from SPIFFS to LittleFS
> If you're using code that uses SPIFFS, you'll need to replace all mentions of SPIFFS in the `.ino` file with LittleFS (see this commit diff for an example of how I changed "Free Reading Wifi" project: [candideu@b86479b](https://github.com/candideu/pocket-wifi-portal-LittleFS/commit/b86479bb79717ef27a58b3134e4bd8be07d05758))
> 
> ![image](https://private-user-images.githubusercontent.com/55474996/287574379-7df11b21-c7a8-4bac-af78-6f5c6501a357.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1NzczNTIsIm5iZiI6MTcwNTU3NzA1MiwicGF0aCI6Ii81NTQ3NDk5Ni8yODc1NzQzNzktN2RmMTFiMjEtYzdhOC00YmFjLWFmNzgtNmY1YzY1MDFhMzU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE4VDExMjQxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzMTA2YzA0Y2UxYmMxZGZkYzVlNGM4YWZkMDMxOTZlZDE4MDQ5NjgxMTFhOTRhZWJjYWViYjMzMmZjODQyMjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Z6hIdnxpxKqR6DFZDF3J1aPOGmYqIpxKZtS9pye96sQ)
> 
> Here's the converted/ported "Free Reading Wifi" file: https://github.com/candideu/pocket-wifi-portal-LittleFS/blob/main/04_Captive_portal_approach_2/Free_Reading_Wifi/Free_Reading_Wifi.ino
> 
> # Upload files with plugin
> Place your project folder in your Sketchbook, as outlined in page 26 of the zine (screen version).
> 
> Open the .ino file. In the Arduino IDE, hit `[Ctrl]` + `[Shift]` + `[P]` (Windows) or `[⌘]` + `[Shift]` + `[P]` (macOS), then type "Upload LittleFS to Pico/ESP8266". Select the option.
> 
> ![image](https://private-user-images.githubusercontent.com/55474996/287575482-0bb603ee-872b-4808-99bf-c5d19f35bb86.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1NzczNTIsIm5iZiI6MTcwNTU3NzA1MiwicGF0aCI6Ii81NTQ3NDk5Ni8yODc1NzU0ODItMGJiNjAzZWUtODcyYi00ODA4LTk5YmYtYzVkMTlmMzViYjg2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE4VDExMjQxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1NWUyZjkwZDg2MjcyYTJhN2U3OTc4ZTgxM2M0ZmZhODBiODhjMjFhNDUzN2E1NmRjYjkxM2I0NWE3ODhhMDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gFP0Txh2RNb75hOTa-j0V5w7mip47XTf0eMWMd2V7TA)
> 
> When the upload is finished, you'll see the message **LittleFS upload completed!** beneath the code editor.
> 
> ![image](https://private-user-images.githubusercontent.com/55474996/287576311-5e32a3c2-b021-4f12-a840-7700eda5a2d8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1NzczNTIsIm5iZiI6MTcwNTU3NzA1MiwicGF0aCI6Ii81NTQ3NDk5Ni8yODc1NzYzMTEtNWUzMmEzYzItYjAyMS00ZjEyLWE4NDAtNzcwMGVkYTVhMmQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE4VDExMjQxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1ZTFhYjQ0ZmI2YjU1MDQzNTRiN2Q5YmFhYjg3Y2FkMmQ0MGRjNmVjZWI0MzRiNWIzMWE5Y2Q0NzBhNzcwNjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.J3wESDPavnsuZO4SdQE137UpW81GQkw7UlpJSSqUW3w)
> 
> Click the **Upload** button in the top left to compile and upload the project code to the Wemos D1 mini.
> 
> ![image](https://private-user-images.githubusercontent.com/55474996/287576584-ab5e24e9-0724-4e82-aa2e-e38f7c3602a1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1NzczNTIsIm5iZiI6MTcwNTU3NzA1MiwicGF0aCI6Ii81NTQ3NDk5Ni8yODc1NzY1ODQtYWI1ZTI0ZTktMDcyNC00ZTgyLWFhMmUtZTM4ZjdjMzYwMmExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE4VDExMjQxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZkYmQxODFlZGJhYTU2MWE1NGRlNjM2YzcwZjRkMjRkMTEyYTE0N2VmN2RhOGU0NzY4NjVmZWUyMGU4ZTUwZWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6kYYMfKfTFvZttxSr8wLYcfsLmivofX1SwaPCeAa1Bo)
> 
> When the upload is done, you'll see **Done uploading** beneath the code editor.
> 
> ![image](https://private-user-images.githubusercontent.com/55474996/287576937-c4b72a61-c99e-430c-b139-5e7484ac02d3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1NzczNTIsIm5iZiI6MTcwNTU3NzA1MiwicGF0aCI6Ii81NTQ3NDk5Ni8yODc1NzY5MzctYzRiNzJhNjEtYzk5ZS00MzBjLWIxMzktNWU3NDg0YWMwMmQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE4VDExMjQxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc3M2NhODlkNDAwNjVjMDc0MDc2ODhkMTU0ZDM4ZWI0YWQwNjEzOTc1N2ZjMGFiZTNjZjk1Yzc1MzlhZTZjYmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TloE40fBNfuzU5hz8Hdo_0iT_iyqsrBmRDAetyzWNy4)

You are using littleFs to upload the files. It doesn't soluction the bug that SPIFFS can't upload. While LittleFS is different with SPIFFS.  [Here](https://arduino-esp8266.readthedocs.io/en/latest/filesystem.html#spiffs-and-littlefs) you will know the different. SPIFFS use less space. However LittleFS is faster.
